creator-node/src/components/tracks/TrackTranscodeHandoffManager.js

```diff
@@ -557,6 +549,50 @@ class TrackTranscodeHandoffManager {
 
     return TrackTranscodeHandoffManager.logger
   }
+
+  /**
+   * Wrapper around async-retry API, with no retry on 404. Used to handle backwards compatibility when
+   * an SP has not yet upgraded and the route is not found.
+   *
+   * options described here https://github.com/tim-kos/node-retry#retrytimeoutsoptions
+   * @param {Object} param
+   * @param {Object} param.logger
+   * @param {func} param.asyncFn the fn to asynchronously retry
+   * @param {string} param.asyncFnLabel the task label used to print on retry. used for debugging purposes
+   * @param {Object} param.options optional options. defaults to the params listed below if not explicitly passed in
+   * @param {number} [param.options.factor=2] the exponential factor
+   * @param {number} [param.options.retries=5] the max number of retries. defaulted to 5
+   * @param {number} [param.options.minTimeout=1000] minimum number of ms to wait after first retry. defaulted to 1000ms
+   * @param {number} [param.options.maxTimeout=5000] maximum number of ms between two retries. defaulted to 5000ms
+   * @param {func} [param.options.onRetry] fn that gets called per retry
+   * @returns the fn response if success, or throws an error
+   */
+  static asyncRetryNotOn404({
+    logger,
+    asyncFn: inputAsyncFn,
+    asyncFnLabel,
+    options = {}
+  }) {
+    const asyncFn = async (bail) => {
+      let resp
+      try {
+        resp = await inputAsyncFn()
+      } catch (e) {
+        // If SP 404's, the SP has not upgraded. Bail without retries.
```

Comment:
one thing that i noticed was hurting us earlier was not specifying the service version after which we can deprecate this - would you mind adding a note here for whatever minimum compatible version would be? much easier to go back and deprecate
example - https://github.com/AudiusProject/audius-protocol/blob/vg-test-transcode-lb-stage/creator-node/src/middlewares.js#L58

also maybe we can add a backlog card to the load balancing project to remove this code? (i recently discovered that notion allows you to set a due date on a card and it'll send you a push notif at the date, which is sick)